Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/PowerSupplies/DieselEngine.cs

```diff
@@ -940,6 +953,24 @@ public void Initialize()
                 GearBox = new GearBox(this);
                 GearBox.Initialize();
             }
+
+            // It appears that the gearbox is initialised in two different places to cater for Basic and Advanced ENG file configurations(?).
+            // Hence the restore values recovered in gearbox class are being overwritten , and resume was not working
+            // Hence restore values are read as part of the diesel and restored at this point
```

Comment:
This is not how you should proceed:

In the current code, we save a boolean indicating if a gearbox has been saved. If this boolean is true, we create the GearBox object in the Restore function, assign the value, then we initialize.

```
bool gearSaved = inf.ReadBoolean();  // read boolean which indicates gear data was saved

if (gearSaved)
{
    GearBox = new GearBox(this);
    GearBox.Restore(inf);
}
```

You just have to modify lines 951 to 955 to check also is the GearBox is currently null and to Initialize if the GearBox object was created either here or in the Restore function.

Reply:
> 
> You just have to modify lines 951 to 955 to check also is the GearBox is currently null and to Initialize if the GearBox object was created either here or in the Restore function.
>
Thanks for the feedback, and suggestion.

I am afraid that I don't quite understand your suggestion.

If you are able to provide me with a small patch file, I would be happy to incorporate it into my changes if it produces a correct restore.